subhub/customer.py

```diff
@@ -80,16 +80,21 @@ def existing_or_new_customer(
         return create_customer(
             subhub_accouunt, user_id, email, source_token, origin_system, display_name
         )
+    logger.info("existing customer", account=db_account)
     customer_id = db_account.cust_id
     return existing_payment_source(customer_id, source_token)
 
 
 @timed
 def existing_payment_source(customer_id: str, source_token: str) -> stripe.Customer:
     existing_customer = stripe.Customer.retrieve(customer_id)
-    if not existing_customer["sources"]["data"]:
-        existing_customer = stripe.Customer.modify(customer_id, source=source_token)
-        logger.info(f"add source {existing_customer}")
+    logger.info("existing customer", data=existing_customer)
+    if not existing_customer.get("sources"):
+        if not existing_customer.get("deleted"):
+            existing_customer = stripe.Customer.modify(customer_id, source=source_token)
+            logger.info(f"add source {existing_customer}")
```

Comment:
don't use f-string for logging; instead use `existing_customer=existing_customer` as kwargs passed to info function.  like you are doing throughout the rest of this code...

Reply:
Good consistency catch!